nkms_eth/actors.py

```diff
@@ -0,0 +1,263 @@
+from collections import OrderedDict
+from typing import Tuple, List
+
+# from nkms_eth.agents import MinerAgent, PolicyAgent
+from nkms_eth.base import Actor
+
+
+class PolicyArrangement:
+    def __init__(self, author: 'PolicyAuthor', miner: 'Miner', value: int,
+                 periods: int, arrangement_id: bytes=None):
+
+        if arrangement_id is None:
+            self.id = self.__class__._generate_arrangement_id()  # TODO: Generate policy ID
+
+        # The relationship exists between two addresses
+        self.author = author
+        self.policy_agent = author.policy_agent
+
+        self.miner = miner
+
+        # Arrangement value, rate, and duration
+        rate = value // periods
+        self._rate = rate
+
+        self.value = value
+        self.periods = periods  # TODO: datetime -> duration in blocks
+
+        self.is_published = False
+
+    @staticmethod
+    def _generate_arrangement_id(policy_hrac: bytes) -> bytes:
+        pass  # TODO
+
+    def __repr__(self):
+        class_name = self.__class__.__name__
+        r = "{}(client={}, node={})"
+        r = r.format(class_name, self.author, self.miner)
+        return r
+
+    def publish(self, gas_price: int) -> str:
+
+        payload = {'from': self.author.address,
+                   'value': self.value,
+                   'gas_price': gas_price}
+
+        txhash = self.policy_agent.transact(payload).createPolicy(self.id,
```

Comment:
So, `transact` takes a dict, obviously?  Why not have `from`, `value`, and `gas_price` as arguments?

Reply:
I have been entertaining this Idea for a while (and I'd really like to see this change, because it helps prevent transmission of ad-hoc transactions).  Since populus's transact method itself takes a `dict` (and our API has been changing), there are instances of the dict being passed with varying keys depending on the context created by solidity methods chained on to `.transact`.  For example: 

 `self.transact({'from': author.address, 'gas_price': gas_price}).revokePolicy(arrangement_id)`

Reply:
I would agree that arguments are probably better than dict (and even then it can work as `transact(**payload)` if needed)

Reply:
Yes - I think that's good.

Reply:
Of course - `from` is a reserved word, and cannot be used as a kwarg here. Perhaps it can change in the contract also.